### DISCUSSION TITLE
Snapping floating windows in float layout

### DISCUSSION THREAD

**Discussion by OmarAshkar** [qtile/qtile#3001] [Ideas]
Hi,
I was wondering if it is possible to have a feature of snapping floating windows to left and right. This is useful as some software I am using works badly when forced into tiles, so It would be useful to have the floating layout work totally like a floating window manager and provide that simple snapping to the right, left, top and bottom halves of the screen. It would be also easier if moving windows in float layout are done with only mouse without the need of a key modifier as moving windows around in a floating mode is essential here.

Is any of that possible?

**Comment by OmarAshkar** [comment]
By the way, awesomewm do that exactly. Floating layout can be snapped. It is not in python unfortunately.

**Reply by elParaguayo** [reply]
Have a look at #2758

**Reply by OmarAshkar** [reply]
@elParaguayo  Ok that is definitely good. But the most important is to snap by the keyboard.  Like in gnome win+left arrow to snap left, or down up to maximize.

**Reply by elParaguayo** [reply]
I've not looked closely at that code but I wouldn't imagine it would be too difficult to add that additional functionality.

**Reply by OmarAshkar** [reply]
Yes, I think it would be great. But actually this snippet is not working. Has it been merged to the release?

**Reply by elParaguayo** [reply]
No, it's not merged yet but that doesn't stop you from forking the repo and applying the patch yourself.

We'll look at the PR when we have the time.

**Reply by OmarAshkar** [reply]
Thank you for all efforts to make this great wm!
